doc/specs/vulkan/chapters/memory.txt

```diff
@@ -446,9 +446,9 @@ endif::VK_EXT_validation_cache[]
 [[memory-device]]
 == Device Memory
 
-Device memory is memory that is visible to the device, for example the
-contents of opaque images that can: be natively used by the device, or
-uniform buffer objects that reside in on-device memory.
+_Device memory_ is a memory that is visible to the device. For example the
```

Comment:
"memory" is a mass noun, so the article "a" should not be added.

Reply:
I don't think these should be split into two sentences - the example sort of clarifies the first sentence. I'd maybe put a semi-colon between them though, rather than a full stop or a comma as was previously there.

Reply:
>"memory" is a mass noun, so the article "a" should not be added.

Ah. Again, feels a bit naked without some quantifier or specifier... perhaps "is a memory type"?

>I don't think these should be split into two sentences

Right. Comma felt wrong to me. Maybe semi-colon or em-dash then.

Reply:
> Ah. Again, feels a bit naked without some quantifier or specifier... perhaps "is a memory type"?

Saying "type" will confuse it with... well.. memory types. I don't think "...is memory that is visible to the device." is particularly weird though - it's grammatically correct.

I mean honestly, the problem is probably that the sentence is circular:

> Device memory is ... **memory** visible to the **device**

...but I think we need a proper memory model before we can fix that aspect of it. For now the best we can do is make sure it's grammatically accurate and adds some amount of information, I think.